devicetypes/smartthings/zigbee-thermostat.src/zigbee-thermostat.groovy

```diff
@@ -138,47 +151,80 @@ private parseAttrMessage(description) {
-			map.unit = temperatureScale
-		} else if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == COOLING_SETPOINT) {
-			log.debug "COOLING SETPOINT"
-			map.name = "coolingSetpoint"
-			map.value = getTemperature(it.value)
-			map.unit = temperatureScale
-		} else if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == HEATING_SETPOINT) {
-			log.debug "HEATING SETPOINT"
-			map.name = "heatingSetpoint"
-			map.value = getTemperature(it.value)
-			map.unit = temperatureScale
-		} else if (it.cluster == THERMOSTAT_CLUSTER && (it.attribute == THERMOSTAT_MODE || it.attribute == THERMOSTAT_RUNNING_MODE)) {
-			log.debug "MODE"
-			map.name = "thermostatMode"
-			map.value = THERMOSTAT_MODE_MAP[it.value]
-			map.data = [supportedThermostatModes: state.supportedModes]
-		} else if (it.cluster == THERMOSTAT_CLUSTER && it.attribute == THERMOSTAT_RUNNING_STATE) {
-			log.debug "RUNNING STATE"
-			def binValue = extendString(bin(hexToInt(it.value)), 16, '0').reverse()
-			map.name = "thermostatOperatingState"
-			if(binValue[0] == "1") {
-				map.value = "heating"
-			} else if(binValue[1] == "1") {
-				map.value = "cooling"
-			} else {
-				map.value = binValue[2] == "1" ? "fan only" : "idle"
+		if (it.cluster == THERMOSTAT_CLUSTER) {
+			if (it.attribute == LOCAL_TEMPERATURE) {
+				log.debug "TEMP"
+				map.name = "temperature"
+				map.value = getTemperature(it.value)
+				map.unit = temperatureScale
+			} else if (it.attribute == COOLING_SETPOINT) {
+				log.debug "COOLING SETPOINT"
+				map.name = "coolingSetpoint"
+				map.value = getTemperature(it.value)
+				map.unit = temperatureScale
+			} else if (it.attribute == HEATING_SETPOINT) {
+				log.debug "HEATING SETPOINT"
+				map.name = "heatingSetpoint"
+				map.value = getTemperature(it.value)
+				map.unit = temperatureScale
+			} else if (it.attribute == THERMOSTAT_MODE || it.attribute == THERMOSTAT_RUNNING_MODE) {
+				log.debug "MODE"
+				map.name = "thermostatMode"
+				map.value = THERMOSTAT_MODE_MAP[it.value]
+				map.data = [supportedThermostatModes: state.supportedThermostatModes]
+			} else if (it.attribute == THERMOSTAT_RUNNING_STATE) {
+				log.debug "RUNNING STATE"
+				def binValue = extendString(bin(hexToInt(it.value)), 16, '0').reverse()
+				map.name = "thermostatOperatingState"
+				if (binValue[0] == "1") {
+					map.value = "heating"
+				} else if (binValue[1] == "1") {
+					map.value = "cooling"
+				} else {
+					map.value = binValue[2] == "1" ? "fan only" : "idle"
+				}
+			} else if (it.attribute == THERMOSTAT_SYSTEM_CONFIG) {
+				def binValue = extendString(bin(hexToInt(it.value)), 16, '0').reverse()
+				// Use reverse() below so that we match Table 6-12. HVAC System Type Configuration Values in the Zigbee Cluster Library spec
+				// The byte representation is RtL, but string indices are LtR...
+				def cooling = binValue[0..1].reverse()
+				def heating = binValue[2..3].reverse()
+				def heatingType = binValue[4]
+				def heatingFuel = binValue[5]
```

Comment:
this is unused, right?

Reply:
Yeah. I had included it since it was a representation of the data, however, best to not waste the extra memory and instructions.